app/logic/certification.py

```diff
@@ -1,14 +1,68 @@
 from peewee import JOIN, DoesNotExist, Case
-
+from flask import g
+from app.models.event import Event
+from app.models.term import Term
 from app.models.certification import Certification
 from app.models.certificationRequirement import CertificationRequirement
 from app.models.requirementMatch import RequirementMatch
 from app.models.eventParticipant import EventParticipant
+from app.models.user import User
+
+def termsAttended(certification=None, username=None):
```

Comment:
We should handle if username is not provided in a different way. Currently, we'll get an error, as `len(attendance)` will break in some fashion, as attendance won't have been defined if there is no username.

We should probably check for a missing username and exit the function right at the top if it is not provided.

Since we are handling that case, we should also have a test case when the expected parameters are not given, and make sure that the function fails gracefully (or fails the way we would expect)